dynamic_programming/lcs.c

```diff
@@ -0,0 +1,151 @@
+/**
+ * @file
+ * @brief [LCS](https://en.wikipedia.org/wiki/Longest_common_subsequence_problem) algorithm
+ * @details
+ * From Wikipedia: The longest common subsequence (LCS) problem is the problem
+ * of finding the longest subsequence common to all sequences in a set of sequences
+ * (often just two sequences).
+ * @author [Kurtz](https://github.com/itskurtz)
+ * @see dynamic_programming/lcs.c
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
```

Comment:
Please add a one-line description of what the library/header is for (**see the example below**).

```c
#include <stdio.h>     /// for IO operations
#include <assert.h>    /// for assert
```